### PR TITLE
indexer-agent reject unsupported subgraph unless bypassed 

### PR DESCRIPTION
**Pull request by hopeyen**
### Goal
- Add a filter such that indexer agent reject to allocate unsupported subgraphs (not eligible for indexing rewards)
- indexers could decide to bypass the filter if desired

### Update
- add additional boolean field `requireSupported` to `IndexingRules`
-- default set to true, such that agent reject unsupported subgraphs by default
-- new rules are set to true by creation
instructions
- if you still want to allocate, bypass the filter by setting `requireSupported` to false
- if you just want to deploy, set `decisionBasis` to offchain

### CODE REVIEW

File: packages/indexer-agent/src/agent.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
 import {
   Allocation,
   allocationRewardsPool,
+  RewardsPool,
```
Comment (fordN) [inline packages/indexer-agent/src/agent.ts:14]: 🤔 I don't see this used below. Is this import needed?

File: packages/indexer-agent/src/network.ts [hunk 2]
```diff
@@ -714,16 +713,27 @@ export class Network {
...
                   },
                 })
 
                 // Skip the indexing rules checks if the decision basis is 'always' or 'never'
                 if (
-                  deploymentRule.decisionBasis === IndexingDecisionBasis.ALWAYS
+                  deploymentRule?.decisionBasis === IndexingDecisionBasis.ALWAYS
                 ) {
+                  // cannot allocate to unsupported subgraph by default
+                  if (
```
Comment (fordN) [inline packages/indexer-agent/src/network.ts:725]: I think this check should be moved outside of the `deploymentRule?.decisionBasis === IndexingDecisionBasis.ALWAYS` conditional block, so it stops all types of rules from allocating towards `unsupported` subgraphs.

**Review by fordN — CHANGES_REQUESTED** [review]

File: packages/indexer-agent/src/agent.ts (hunk 1 above)
Comment (hopeyen) [reply packages/indexer-agent/src/agent.ts:14]: no idea where it came from 🤨  rebasing and then will remove if still there

Comment (hopeyen) [reply packages/indexer-agent/src/agent.ts]: I removed it and then found there's a reference to it at line 502 `uniqueRewardsPools: RewardsPool[]`. I don't think it belonged in this PR...

Comment (fordN) [reply packages/indexer-agent/src/agent.ts]: Ok, now the changlist shows RewardsPool as removed.  Needs to be added back in and in the future if the PRs could be based off main and rebased on main before review that would make it easier to see what changes are made in this PR.

**Review by fordN — APPROVED** [review]
👍

**Pull request merged**